### PR TITLE
Add status-code/cci.20220616

### PR DESCRIPTION
**Pull request by hnOsmium0001**
Specify library name and version:  **status-code/cci.20220616**

- A reference implementation for a standard paper, potentially useful
- Dependency of https://github.com/ned14/llfio
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by hnOsmium0001** [comment]
Well it seems like their README lied
> ortable to any C++ 11 compiler. These are known to work:
> - \>= GCC 5 (due to requiring libstdc++ 5 for sufficient C++ 11 type traits)

File: recipes/status-code/all/test_package/test_package.cpp [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#include <status_code.hpp>
+
+int main() {
```
Comment (prince-chrismc) [inline recipes/status-code/all/test_package/test_package.cpp:3]: maybe add a one liner for sanity?

File: recipes/status-code/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+find_package(status-code REQUIRED CONFIG)
+
+add_executable(test_package test_package.cpp)
+target_link_libraries(test_package status-code::hl)
+target_compile_features(test_package PRIVATE cxx_std_11)
+
+enable_testing()
+add_test(NAME test_package
+         COMMAND test_package)
```
Comment (prince-chrismc) [inline recipes/status-code/all/test_package/CMakeLists.txt:12]: not required for CCI

```suggestion
target_compile_features(test_package PRIVATE cxx_std_11)
```

File: recipes/status-code/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                  destination=self._source_subfolder, strip_root=True)
+
+    def package(self):
+        self.copy("Licence.txt", dst="licenses", src=self._source_subfolder)
+        self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, "include"))
+        self.copy("*.ipp", dst="include", src=os.path.join(self._source_subfolder, "include"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "status-code")
+        self.cpp_info.set_property("cmake_target_name", "status-code::hl")
```
Comment (prince-chrismc) [inline recipes/status-code/all/conanfile.py:53]: 👏 https://github.com/ned14/status-code/blob/38e1e862386cb38d577664fd681ef829b0e03fba/CMakeLists.txt#L126

File: recipes/status-code/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    def _compiler_required_cpp17(self):
+        return {
+            # Their README says gcc 5, but in testing only >=7 worked
+            "gcc": "7",
+            "clang": "3.3",
+            "Visual Studio": "14",
+            "apple-clang": "5",
+        }
+
+    def validate(self):
```
Comment (prince-chrismc) [inline recipes/status-code/all/conanfile.py:31]: I think you are missing one thing here too https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-is-a-toolscheck_min_cppstd-call-not-enough

File: recipes/status-code/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, _cppstd_required_version)
```
Comment (prince-chrismc) [inline recipes/status-code/all/conanfile.py:37]:
```suggestion
            tools.check_min_cppstd(self, self._cppstd_required_version)
```

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
cla is missing

**Review by SSE4 — APPROVED** [review]

**Pull request merged**